datafusion/common/src/scalar.rs

```diff
@@ -1970,13 +2020,14 @@ impl ScalarValue {
                 ),
             },
             ScalarValue::Fixedsizelist(..) => {
-                unimplemented!("FixedSizeList is not supported yet")
```

Comment:
What would be the preferred way to handle unimplemented errors in `datafusion`? There are many places where a `NotImplemented` error is returned instead of using `unimplemented!` and panicking. IMO returning an error makes more sense as user can choose to ignore unimplemented errors instead of panicking and exiting.